docs/root/configuration/http_filters/tap_filter.rst

```diff
@@ -0,0 +1,122 @@
+    admin_config:
+      config_id: test_config_id
+
+The previous snippet configures the filter for control via the :http:post:`/tap` admin handler.
+See the following section for more details.
+
+.. _config_http_filters_tap_admin_handler:
+
+Admin handler
+-------------
+
+When the HTTP filter specifies an :ref:`admin_config
+<envoy_api_msg_config.filter.http.tap.v2alpha.AdminConfig>`, it is configured for admin control and
+the :http:post:`/tap` admin handler will be installed. The admin handler can be used for live
+tapping and debugging of HTTP traffic. It works as follows:
+
+1. A POST request is used to provide a valid tap configuration. The POST request body can be either
+   the JSON or YAML representation of the :ref:`TapConfig
+   <envoy_api_msg_service.tap.v2alpha.TapConfig>` message.
+2. If the POST request is accepted, Envoy will stream :ref:`HttpBufferedTrace
+   <envoy_api_msg_data.tap.v2alpha.HttpBufferedTrace>` messages (serialized to JSON) until the admin
+   request is terminated.
+
+An example POST body:
+
+.. code-block:: yaml
+
+  config_id: test_config_id
+  tap_config:
+    match_configs:
+      - match_id: foo_match_id
+        http_match_config:
+          request_match_config:
+            headers:
+              - name: foo
+                exact_match: bar
+          response_match_config:
+            headers:
+              - name: bar
+                exact_match: baz
+    output_config:
+      sinks:
+        - streaming_admin: {}
+
+The configuration instructs the tap filter to match any HTTP requests in which a request header
+``foo: bar`` is present AND a response header ``bar: baz`` is present. If both of these conditions
+are met, the request will be tapped and streamed out the admin endpoint.
+
+Another example POST body:
+
+.. code-block:: yaml
+
+  config_id: test_config_id
+  tap_config:
+    match_configs:
+      - match_id: request_match_id
+        http_match_config:
+          request_match_config:
+            headers:
+              - name: foo
+                exact_match: bar
+      - match_id: response_match_id
+        http_match_config:
+          response_match_config:
+            headers:
+              - name: bar
+                exact_match: baz
+    output_config:
+      sinks:
+        - streaming_admin: {}
+
+The configuration instructs the tap filter to match any HTTP requests in which a request header
+``foo: bar`` is present OR a response header ``bar: baz`` is present. If either of these conditions
+are met, the request will be tapped and streamed out the admin endpoint. The difference between
+the first example and the second is the use of multiple top level match configurations to produce
+a logical OR.
+
+Statistics
+----------
+
+The tap filter outputs statistics in the *http.<stat_prefix>.tap.* namespace. The :ref:`stat prefix
+<envoy_api_field_config.filter.network.http_connection_manager.v2.HttpConnectionManager.stat_prefix>`
+comes from the owning HTTP connection manager.
```

Comment:
Would it be also useful to include the opaque ID here; even if strictly inferable, it seems a helpful hint.

Reply:
I thought about this. I'm not sure if it's useful or not. I'm going to put in a TODO and we can see what kind of stats feedback we get.